src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6605,6 +6605,13 @@ public void ClearContent(string path)
 
             path = NormalizePath(path);
 
+            if (Directory.Exists(path))
```

Comment:
Formally no need to do the path normalization before `Directory.Exists`. We could place the addition before line 6606.

Reply:
@iSazonov Do you want me to put it before NormalizePath() ?

Reply:
Yes, I think it make sense.

Reply:
sure, back to square 1

Reply:
@iSazonov `NormalizePath()` fixes backslashes/forwardslashes.  Why wouldn't we normalize it before checking if the dir exists?  On macOS, [Directory]::Exists("\Users\steve") fails because of the slashes.

Reply:
Very valid point 👍

Reply:
@SteveL-MSFT Thanks! I see that CoreFX always normalize paths only on Windows and keep '\' "as is" because it is valid path char.

Closed.

Reply:
@iSazonov I see what you are saying.  It seems the right thing to do here then is see if `-LiteralPath` was used and if so, don't normalize.

Reply:
Just checked and even `Set-Content -LiteralPath` doesn't handle backslashes correctly on non-Windows as it normalizes the path first.  We can consider this out of scope for this PR.

Reply:
@SteveL-MSFT Could you please open new issue for `Set-Content -LiteralPath` problem? I still don't understand how wide in the code base is the problem.

Reply:
https://github.com/PowerShell/PowerShell/issues/8158